File: 02_Object_Oriented_Programming/02_Class_Relationships/03_Aggregation/javascript/player.js
/**
 * Aggregation — Player class in JavaScript.
 */

class Player {
  #name;
  #jerseyNumber;

  constructor(name, jerseyNumber) {
    this.#name = name;
    this.#jerseyNumber = jerseyNumber;
  }

  get name() {
    return this.#name;
  }

  get jerseyNumber() {
    return this.#jerseyNumber;
  }

  toString() {
    return `${this.#name} (#${this.#jerseyNumber})`;
  }
}

module.exports = Player;

File: 02_Object_Oriented_Programming/02_Class_Relationships/03_Aggregation/java/Player.java
/**
 * Aggregation — Player class.
 *
 * Key concept: A Team HAS Players, but Players can exist independently.
 * If the Team is destroyed, Players still exist (weak ownership).
 */
public class Player {

    private String name;
    private int jerseyNumber;

    public Player(String name, int jerseyNumber) {
        this.name = name;
        this.jerseyNumber = jerseyNumber;
    }

    public String getName() {
        return name;
    }

    public int getJerseyNumber() {
        return jerseyNumber;
    }

    @Override
    public String toString() {
        return name + " (#" + jerseyNumber + ")";
    }
}
